src/openrct2/object/ObjectRepository.cpp

```diff
@@ -98,13 +98,12 @@ class ObjectRepository final : public IObjectRepository
     std::vector<ObjectRepositoryItem>   _items;
     QueryDirectoryResult                _queryDirectoryResult = { 0 };
     ObjectEntryMap                      _itemMap;
-    uint16                              _languageId = 0;
-    sint32                                 _numConflicts;
+    uint16                              _languageId   = 0;
+    sint32                              _numConflicts = 0;
 
 public:
-    ObjectRepository(IPlatformEnvironment * env)
+    ObjectRepository(IPlatformEnvironment * env) : _env(env)
     {
-        _env = env;
     }
```

Comment:
> any read-only field can be set providing it's in the constructor.

Which effectively violates its constness, as far as I understand that. Such fields are guaranteed to be initialised *before* the ctor starts executing in C++, but C# seems to discard that.